zucchini-ui-frontend/src/stepDefinitions/components/StepDefinitionsTable.jsx

```diff
@@ -0,0 +1,255 @@
+        <thead>
+          <tr>
+            <th className='col-md-10'>Définition</th>
+            <th className='col-md-1' style={{ textAlign: 'center' }}>Occurrence</th>
+            <th className='col-md-1' style={{ textAlign: 'center' }}>Réussite</th>
+          </tr>
+        </thead>
+        <tbody>{rows}</tbody>
+      </Table>
+    );
+  }
+}
+
+class StepDefinitionsRow extends React.Component {
+
+  static propTypes = {
+    stepDefinition: PropTypes.object.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      showVariants: false,
+    };
+  }
+
+  onShowVariants = () => {
+    this.setState({
+      showVariants: true,
+    });
+  };
+
+  onHideVariants = () => {
+    this.setState({
+      showVariants: false,
+    });
+  };
+
+  render() {
+
+    const { stepDefinition } = this.props;
+    const stepLocation = `${stepDefinition.stepDefinitionLocation.filename}:${stepDefinition.stepDefinitionLocation.line}`;
+
+    const nbSuccesses = stepDefinition.occurrences.filter((step) => {
+      return step.status === 'PASSED'
+    }).length;
+    const successRate = Math.floor((nbSuccesses / stepDefinition.occurrences.length) * 100);
+
+    let successBadge;
+    if (successRate >= 90) {
+      successBadge = 'badge-success';
+    } else if (successRate >= 50 && successRate >= 90) {
+      successBadge = 'badge-warning';
+    } else if (successRate < 50) {
+      successBadge = 'badge-error';
+    }
+
+    return (
+      <tr key={stepDefinition}>
+        <td style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>
+          <a onClick={this.onShowVariants}>
+            <StepDefinitionsCell occurrence={stepDefinition.occurrences[0]} />
+          </a>
+          <StepDefinitionsVariantsDialog occurrences={stepDefinition.occurrences} location={stepLocation} show={this.state.showVariants} onClose={this.onHideVariants} />
+        </td>
+        <td style={{ textAlign: 'center' }}>
+          <b>{stepDefinition.occurrences.length}</b>
+        </td>
+        <td style={{ textAlign: 'center' }}>
+          <Badge bsClass={successBadge}>{successRate} %</Badge>
+        </td>
+      </tr>
+    );
+  }
+}
+
+class StepDefinitionsCell extends React.Component {
+
+  static propTypes = {
+    occurrence: PropTypes.object.isRequired
+  };
+
+  render() {
+    const { occurrence } = this.props;
+    const highlightedText = formatStepDefinition(occurrence);
+    return (
+      <span>{highlightedText.map((text, idx) => <span key={idx}>{text}</span>)}</span>
+    );
+  }
+}
+
+class StepDefinitionsTableHighlightedTerm extends React.Component {
+
+  static propTypes = {
+    text: PropTypes.string.isRequired,
+    styleClass: PropTypes.string.isRequired,
+  };
+
+  render() {
+    const { text, styleClass } = this.props;
+    return (
+      <span className={styleClass}>{text}</span>
+    );
+  }
+}
+
+class StepDefinitionsVariantsDialog extends React.PureComponent {
```

Comment:
The StepDefinitionsVariantsDialog should be in its own file (with its functions).

Reply:
OK. Extracted to its file